homeassistant/components/media_player/plex.py

```diff
@@ -272,93 +527,383 @@ def _active_media_plexapi_type(self):
 
     @property
     def media_image_url(self):
         """Image url of current playing media."""
-        if self.session is not None:
-            thumb_url = self._convert_na_to_none(self.session.thumbUrl)
-            if str(self.na_type) in thumb_url:
-                # Audio tracks build their thumb urls internally before passing
-                # back a URL with the PlexAPI _NA type already converted to a
-                # string and embedded into a malformed URL
-                thumb_url = None
-            return thumb_url
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.session:
+                thumb_url = self._convert_na_to_none(self.session.thumb)
+                if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                    if not self.optional_config[CONF_USE_EPISODE_ART]:
+                        thumb_url = self._convert_na_to_none(
+                            self.session.grandparentThumb)
+
+                if thumb_url:
+                    thumb_url = self.session.server.url(thumb_url)
+                    thumb_response = requests.get(thumb_url, verify=False)
+                    if thumb_response.status_code != 200:
+                        _LOGGER.debug('Using media art because media thumb '
+                                      'was not found: %s', self.entity_id)
+                        thumb_url = self.session.server.url(
+                            self._convert_na_to_none(self.session.art))
+                else:
+                    thumb_url = self.session.server.url(
+                        self._convert_na_to_none(self.session.art))
+
+                return thumb_url
 
     @property
     def media_title(self):
         """Title of current playing media."""
-        # find a string we can use as a title
-        if self.session is not None:
-            return self._convert_na_to_none(self.session.title)
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            # find a string we can use as a title
+            title = None
+            if self.session:
+                if self._convert_na_to_none(self.session.title) is not None:
+                    title = self.session.title
+
+                    # append year for movies
+                    if self.media_content_type is MEDIA_TYPE_VIDEO:
+                        if self._convert_na_to_none(
+                                self.session.year) is not None:
+                            title = title + " (" + str(self.session.year) + ")"
+
+            return title
 
     @property
     def media_season(self):
         """Season of curent playing media (TV Show only)."""
-        return self._season
+        if self.state not in (STATE_IDLE, STATE_OFF):  # true or frozen state
+            if self.media_content_type is MEDIA_TYPE_TVSHOW:
+                if self.session is not None:
+                    if callable(self.session):
```

Comment:
Session is not used as a callable anywhere else, are you sure this should not be `seasons`?